sqlite_fdw.c

```diff
@@ -5690,3 +5718,279 @@ sqliteIsForeignRelUpdatable(Relation rel)
+		TupleDesc	tupdesc = RelationGetDescr(rel);
+
+		relname = RelationGetRelationName(rel);
+		if (errpos->cur_attno > 0 && errpos->cur_attno <= tupdesc->natts)
+		{
+			Form_pg_attribute attr = TupleDescAttr(tupdesc,
+												   errpos->cur_attno - 1);
+
+			attname = NameStr(attr->attname);
+		}
+		else if (errpos->cur_attno == SelfItemPointerAttributeNumber)
+			attname = "ctid";
+	}
+
+	{
+		/*
+		 * Error HINT block
+		 */
+		char	   *err_hint_mess0 = palloc(max_logged_byte_length * 2 + 1024); /* The longest hint message */
+		char 	   *err_hint_mess;
+		char	   *value_text = NULL;
+		bool		sqlite_value_as_hex_code = value_byte_size_blob_or_utf8 < max_logged_byte_length && ((GetDatabaseEncoding() != PG_UTF8 && value_aff == SQLITE3_TEXT) || (value_aff == SQLITE_BLOB));
+
+		/* Print problem SQLite values only for
+		 * - integer,
+		 * - float,
+		 * - short text if database encoding is UTF-8
+		 *   incorrect output otherwise possible: UTF-8 in SQLite, but not supported charcters in PostgreSQL
+		 */
+		if ((value_byte_size_blob_or_utf8 < max_logged_byte_length && GetDatabaseEncoding() == PG_UTF8 && value_aff == SQLITE3_TEXT)
+			|| value_aff == SQLITE_INTEGER
+			|| value_aff == SQLITE_FLOAT)
+			value_text = (char *)sqlite3_value_text(errpos->val);
+
+		if (sqlite_value_as_hex_code)
+		{
+			const unsigned char *vt = sqlite3_value_text(errpos->val);
+			value_text = palloc (max_logged_byte_length * 2 + 1);
+			for (size_t i = 0; i < value_byte_size_blob_or_utf8; ++i)
+				sprintf(value_text + i * 2, "%02x", vt[i]);
+	    }
+
+		err_hint_mess = err_hint_mess0;
+		if (value_text != NULL)
+		{
+			if (sqlite_value_as_hex_code)
+				err_hint_mess += sprintf(
+						err_hint_mess,
+						"SQLite hex value %s\n",
+						value_text );
+			else if (value_aff != SQLITE_INTEGER && value_aff != SQLITE_FLOAT)
+				err_hint_mess += sprintf(
+						err_hint_mess,
+						"SQLite value '%s'\n",
+						value_text );
+			else
+				err_hint_mess += sprintf(
+						err_hint_mess,
+						"SQLite value %s\n",
+						value_text );
+		err_hint_mess += sprintf(
+			err_hint_mess,
+			"("
+			);
+		}
+		err_hint_mess += sprintf(
+			err_hint_mess,
+			"affinity \"%s\"",
+			sqlite_affinity
+			);
+		if (value_aff == SQLITE3_TEXT || value_aff == SQLITE_BLOB )
+			err_hint_mess += sprintf(
+					err_hint_mess,
+					", length %d bytes",
+					value_byte_size_blob_or_utf8 );
+		if (value_text != NULL)
+			err_hint_mess += sprintf(err_hint_mess, ")");
+
+		err_hint_mess[1] = '\0';
+		errhint("%s", err_hint_mess0);
+		pfree(err_hint_mess0);
+		if (sqlite_value_as_hex_code)
+			pfree((char *)value_text);
+	}
+
+	{
+		/*
+		 * Error CONTEXT block
+		 */
+		char	   *err_cont_mess0 = palloc(4 * NAMEDATALEN + 64); /* The longest context message */
+		char 	   *err_cont_mess;
+
+		err_cont_mess = err_cont_mess0;
+		err_cont_mess = err_cont_mess + sprintf(
+			err_cont_mess,
+			"foreign table \"%s\"\nforeign column \"%.*s\" have data type \"%s\" (usual affinity \"%s\")\n",
```

Comment:
How do you think about message: `"column \"%.*s\" of foreign table \"%s\" have data type \"%s\" (usual affinity \"%s\")"`

Reply:
I am agree. Really `\n` is not necessary in `CONTEXT` part of error message.

Reply:
Also additional words about query from reference `postgres_fdw` implementation added without `\n` in https://github.com/pgspider/sqlite_fdw/pull/91/commits/311b818fdde67cd0bc5b45ac600d7359cfcc6d6f. Please verify, @nxhai98.

Reply:
Thank you. Please help me resolve this thread.